### PR TITLE
Add generation of client code

### PR DESCRIPTION
**Pull request by mgianatagh**
- Add generation of client code for the verrazzano custom resource
- Remove some Makefile dependencies that were automatically generating code

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,12 +98,26 @@ go-mod:
 	$(GO) mod tidy
 	$(GO) mod vendor
 
+	# go mod vendor only copies the .go files.  Also need
+	# to populate the k8s.io/code-generator folder with the
+	# scripts for generating informer/lister code
+
+	# Obtain k8s.io/code-generator version
+	$(eval codeGenVer=$(shell grep "code-generator" go.mod | awk '{print $$2}'))
```
Comment (markxnelson) [inline Makefile:106]: we do not want makefiles to automatically generate code - see decision log

**Review by markxnelson — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (pfmackin) [reply Makefile:106]: this client gen should be under the generate target

Comment (mgianatagh) [reply Makefile:106]: this it not generating code, it is populating the vendor folder

However, while reviewing your comment I noticed the target `docker-build` has an pre-existing dependency on `generate` and `manifests`.  I will remove those.

Comment (mgianatagh) [reply Makefile:106]: I pushed a change that removes the pre-existing dependencies on targets that generate code

**Review by markxnelson — APPROVED** [review]

**Review by pfmackin — APPROVED** [review]

**Pull request merged**